Crud

niktimmans opened the pull request.

maria-kanyushkova committed:
add list View for book api

maria-kanyushkova committed:
update book models and serializers

niktimmans committed:
CRUD
-elasticsearch added
-made crud for book
need:
pip install elasticsearch
pip install django-elasticsearch
issues:
might be problem with import error for this change "FileDoesNotExist" import to "ObjectDoesNotExist" in managers.py and django_elasticsearch\serializers.py

maria-kanyushkova commented on book/migrations/0006_auto_20201113_2106.py:
```diff
@@ -15,7 +15,7 @@ def addBooks(apps, schema_editor):
 def removeBooks(apps, schema_editor):
     Book = apps.get_model("book", "book")
 
```
Не надо было это править, это именно удалить добавленные с помощью миграции тестовые данные

maria-kanyushkova commented on book/models.py:
```diff
@@ -10,19 +11,19 @@ class Asset(models.Model):
 
 
-class Author(models.Model):
+class Author(EsIndexable, models.Model):
```
EsIndexable Это для чего?

maria-kanyushkova reviewed (commented):
Нужны ответы, и хочу проверить работоспособность завтра

maria-kanyushkova commented:
нужно еще в сеттингсы добавить ELASTICSEARCH_AUTO_INDEX = True
для автоиндексации

maria-kanyushkova commented on book/views.py:
```diff
@@ -1,9 +1,24 @@
-    b = Book(title="title", description="description", rate="0", isbn="isbn")
-    b.save()
-    return HttpResponse("Hello, world. You're at the books index.")
+# def index(request):
+#     # b = Book(title="title", description="description", rate="0", isbn="isbn")
+#     # b.save()
+#     return HttpResponse("Hello, world. You're at the books index.")
```
нужно будет выпилить

niktimmans committed:
new:
-required fields are now optional
-elasticsearch auto index = true added
-unused comment deleted
